src/main/kotlin/features/items/EtherwarpOverlay.kt

```diff
@@ -0,0 +1,53 @@
+package moe.nea.firmament.features.items
+
+import moe.nea.firmament.annotations.Subscribe
+import moe.nea.firmament.features.FirmamentFeature
+import moe.nea.firmament.gui.config.ManagedConfig
+import moe.nea.firmament.util.MC
+import net.minecraft.util.hit.BlockHitResult
+import moe.nea.firmament.events.WorldRenderLastEvent
+import moe.nea.firmament.util.extraAttributes
+import moe.nea.firmament.util.render.RenderInWorldContext
+import moe.nea.firmament.util.skyBlockId
+
+object EtherwarpOverlay : FirmamentFeature {
+	override val identifier: String
+		get() = "etherwarp-overlay"
+
+	object TConfig : ManagedConfig(identifier, Category.ITEMS) {
+		var etherwarpOverlay by toggle("etherwarp-overlay") { false }
+		var cube by toggle("cube") { true }
+		var wireframe by toggle("wireframe") { false }
+	}
+
+	override val config: ManagedConfig
+		get() = TConfig
+
+
+	@Subscribe
+	fun renderEtherwarpOverlay(event: WorldRenderLastEvent) {
+		val player =
+			MC.player?.takeIf { MC.world != null && MC.camera != null && TConfig.etherwarpOverlay && it.isSneaking }
+				?: return
+		player.mainHandStack.skyBlockId?.takeIf { it.neuItem == "ASPECT_OF_THE_VOID" || it.neuItem == "ASPECT_OF_THE_END" }
+			?: return
+		if (player.mainHandStack.extraAttributes.get("ethermerge") == null) return
+		val camera = MC.camera ?: return
+		val world = MC.world ?: return
```

Comment:
Please split this up
```suggestion
		if (!TConfig.etherwarpOverlay) return
		val player = MC.player ?: return
		val world = player.world
		val camera = MC.camera ?: return
		val heldItem = MC.stackInHand
		if (heldItem.skyBlockId !in listOf(SkyBlockItems.ASPECT_OF_THE_VOID, SkyBlockItems.ASPECT_OF_THE_END)) return
		if (!heldItem.extraAttributes.contains("ethermerge")) return
```